src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/CompatDifference.cs

```diff
@@ -65,5 +66,22 @@ public CompatDifference(string diagnosticId, string message, DifferenceType type
         /// </summary>
         /// <returns><see cref="string"/> describing the difference.</returns>
         public override string ToString() => $"{DiagnosticId} : {Message}";
+
+        public bool Equals(CompatDifference other) => other != null &&
```

Comment:
Do you also need `public bool override Equals(object other)`?

Reply:
It wouldn't be called by HashCode<T> since the (default) GenericEqualityComparer<T> will always call the concrete one in our case, but it also doesn't hurt to add so I can go ahead and add it.

Reply:
I see.  I usually follow this rule religiously as I've had to debug plenty of the times it's been broken:
https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-5.0#remarks
> If you override the GetHashCode method, you should also override Equals, and vice versa. If your overridden Equals method returns true when two objects are tested for equality, your overridden GetHashCode method must return the same value for the two objects.